### PR TITLE
Use vm_common set of core libraries in libraries.yaml files

### PR DESCRIPTION
**Pull request by alexmarkov**
This is a refactoring of libraries.yaml / libraries.json to reference `vm_common` set of libraries defined in Dart SDK (https://dart-review.googlesource.com/c/sdk/+/260900).

The goal is to have one source of truth and simplify future changes in the core library patch files across all embeddings of Dart VM (standalone Dart VM, Flutter, Fuchsia Dart runner and Fuchsia Flutter runner).

### CODE REVIEW

File: lib/snapshot/libraries.yaml [hunk 1]
```diff
@@ -2,158 +2,31 @@
...
-    nativewrappers:
-      uri: "../../../third_party/dart/sdk/lib/html/dartium/nativewrappers.dart"
-
-    profiler:
-      uri: "../../../third_party/dart/sdk/lib/profiler/profiler.dart"
-
-    typed_data:
-      uri: "../../../third_party/dart/sdk/lib/typed_data/typed_data.dart"
-      patches: "../../../third_party/dart/sdk/lib/_internal/vm/lib/typed_data_patch.dart"
+    _spirv:
```
Comment (johnniwinther) [inline lib/snapshot/libraries.yaml:26]: So this was wrongfully missing from the .yaml version before, right?

File: shell/platform/fuchsia/flutter/kernel/libraries.yaml [hunk 2]
```diff
@@ -161,11 +35,10 @@ flutter_runner:
     fuchsia:
       uri: "../../dart-pkg/fuchsia/lib/fuchsia.dart"
 
+    _spirv:
```
Comment (johnniwinther) [inline shell/platform/fuchsia/flutter/kernel/libraries.yaml:38]: And similar here, right?

**Review by johnniwinther — APPROVED** [review]

File: lib/snapshot/libraries.yaml (hunk 1 above)
Comment (alexmarkov) [reply lib/snapshot/libraries.yaml:26]: Yes. libraries.json for Flutter and Flutter runner include "_spirv" library but corresponding libraries.yaml don't. As far as I understand CFE uses libraries.json, so "_spirv" library is in fact included into the core libraries. So I updated libraries.yaml to reflect that and re-generated libraries.json.

File: shell/platform/fuchsia/flutter/kernel/libraries.yaml (hunk 2 above)
Comment (alexmarkov) [reply shell/platform/fuchsia/flutter/kernel/libraries.yaml:38]: Yep.

File: lib/snapshot/libraries.yaml (hunk 1 above)
Comment (jonahwilliams) [reply lib/snapshot/libraries.yaml:26]: _spirv library should have been deleted when we removed the transpiler. It's possible we missed something though

Comment (jonahwilliams) [reply lib/snapshot/libraries.yaml:26]: Yes, I believe all of the _spirv imports are removed. Fyi @zanderso

Comment (alexmarkov) [reply lib/snapshot/libraries.yaml]: I checked other Flutter core libraries and found that in addition to '_spirv' there are also obsolete 'wasm' and 'profiler' libraries which reference non-existing .dart files. Cleaned up all of them, PTAL.

Comment (zanderso) [reply lib/snapshot/libraries.yaml]: Correct. This reminded me to do this https://github.com/flutter/engine/pull/36422...

**Review by jonahwilliams — APPROVED** [review]
Lgtm

**Pull request merged**